### PR TITLE
Bugfix: ZipFS and TarFS should not have a syspath.

### PR DESCRIPTION
**Pull request by tfeldmann**
## Type of changes

- Bug fix
- Tests

## Checklist

- [x] I've run the latest [black](https://github.com/ambv/black) with default args on new code.
- [x] I've updated CHANGELOG.md and CONTRIBUTORS.md where appropriate.
- [x] I've added tests for new code.
- [x] I accept that @PyFilesystem/maintainers may be pedantic in the code review.

## Description

Fixes Issue #528.

### CODE REVIEW

**Comment by tfeldmann** [comment]
Oops,I forgot to fetch upstream before the implementation - the changelog has a merge conflict.

File: tests/test_tarfs.py [hunk 1]
```diff
@@ -64,6 +64,10 @@ def destroy_fs(self, fs):
         os.remove(fs._tar_file)
         del fs._tar_file
 
+    def test_nosyspath(self):
+        with self.assertRaises(NoSysPath):
+            self.fs.getsyspath("/test")
```
Comment (lurch) [inline tests/test_tarfs.py:69]: Just to err on the side of caution, is it worth checking `getospath` too?

**Comment by lurch** [comment]
Apart from the still-present merge conflict, LGTM.

**Comment by lurch** [comment]
This PR adds 4 lots of `def getsyspath` but only 3 lots of `def hassyspath` ?

**Comment by tfeldmann** [comment]
Fixed that, although `hassyspath` already returned False for ReadTarFS. I guess it's clearer this way.
Please do not merge yet, this PR needs some further discussion in #528.

**Comment by tfeldmann** [comment]
Closing this as a non-issue, see #528

**Pull request closed without merging**